### PR TITLE
chore: refactoring README and DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by suztomo**
Thank you for opening a Pull Request! For general contributing guidelines, please refer to [contributing guide](https://github.com/googleapis/gapic-generator-java/blob/main/CONTRIBUTING.md)

- The old file https://github.com/googleapis/gapic-generator-java/blob/main/gapic-generator-java/DEVELOPMENT.md was last touched in mid December. I'm replacing the content with the latest at the root.
- Removing the root DEVELOPMENT.md in favor of the file above.
- Adding index of the modules at the root README.md
- Moving showcase testing guide to showcase/README.md.

### CODE REVIEW

File: gapic-generator-java/DEVELOPMENT.md [hunk 1]
```diff
@@ -2,209 +2,172 @@
...
     ```sh
     cp .githooks/pre-commit .git/hooks/pre-commit
     ```
 
+3. Install [`bazelisk`](https://github.com/bazelbuild/bazelisk) in your `PATH`.
+
 ## Code Formatting
 
--   Run linter checks without actually doing the formatting.
+- Run linter checks without actually doing the formatting.
```
Comment (suztomo) [inline gapic-generator-java/DEVELOPMENT.md:19]: They come from the latest https://github.com/googleapis/gapic-generator-java/blob/main/DEVELOPMENT.md (at the root).

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -1,261 +0,0 @@
-# Development Workflow
-
-## Set Up
```
Comment (blakeli0) [inline DEVELOPMENT.md:3]: Can we still keep some of the info in the root DEVELOPMENT.md? For example, the set up, code formatting, and running integration tests.

Comment (suztomo) [reply DEVELOPMENT.md:3]: Done. Yes, the pre-commit hooks and format should be at the root.

File: gapic-generator-java/DEVELOPMENT.md [hunk 3]
```diff
@@ -1,210 +1,145 @@
...
-
 ## Test Running
 
--   Run all unit and integration tests.
+- Run all unit and integration tests.
 
     ```sh
-    bazel test //...
+    mvn install          # unit tests, maven test wouldn't work in root folder because gapic-generator-java is dependant on test jars of gax-java
+    bazel test //...  # integration tests
```
Comment (blakeli0) [inline gapic-generator-java/DEVELOPMENT.md:9]: The integration tests can only be run from the root folder, and integration tests can also used for changes in gax, so we may want to move this line and references of integration tests to root folder.

File: gapic-generator-java/DEVELOPMENT.md [hunk 4]
```diff
@@ -1,210 +1,145 @@
...
-    files in `test/integration/goldens/redis`.
+- Update integration test golden files, for example `Redis`. This clobbers all the
+  files in `test/integration/goldens/redis`.
 
     ```sh
     bazel run //test/integration:update_redis
     ```
 
-## Running the Plugin
+## Running the Plugin under googleapis with local gapic-generator-java
```
Comment (blakeli0) [inline gapic-generator-java/DEVELOPMENT.md:47]: I think we could move this whole section to root folder also, as we can use this way to test local gax changes as well.

File: gapic-generator-java/DEVELOPMENT.md (hunk 3 above)
Comment (suztomo) [reply gapic-generator-java/DEVELOPMENT.md]: Because this integration test is for gapic-generator-java submodule, I keep it here noting that developers should run these commands in the root directory of the repository.

File: gapic-generator-java/DEVELOPMENT.md (hunk 4 above)
Comment (suztomo) [reply gapic-generator-java/DEVELOPMENT.md:76]: I prefer keeping DEVELOPMENT.md specific to module. To address test local gax change, I created gax-java/DEVELOPMENT.md that points to gapic-generator-java/DEVELOPMENT.md for local testing.

File: gax-java/CONTRIBUTING.md [hunk 5]
```diff
@@ -20,21 +20,6 @@ frustration later on.
 All submissions, including submissions by project members, require review. We
 use Github pull requests for this purpose.
 
-### Building
```
Comment (suztomo) [inline gax-java/CONTRIBUTING.md:23]: This moves to gax-java/DEVELOPMENT.md

**Review by suztomo — COMMENTED** [review]
@blakeli0 PTAL

File: gax-java/README.md [hunk 6]
```diff
@@ -15,7 +15,7 @@ files can use services such as paged list iteration, request batching, and
 polling of long-running operations to provide a more convenient and idiomatic
 API surface to callers.
 
-Currently, this library shouldn't be used independently from google-cloud-java, otherwise there is
+Currently, this library shouldn't be used independently of google-cloud-java, otherwise there is
 a high risk of diamond dependency problems, because google-cloud-java uses beta features from this
 library which can change in breaking ways between versions. See [VERSIONING](#versioning) for
 more information.
```
Comment (blakeli0) [inline gax-java/README.md:21]: Noticed that some other content of this readme need to be updated too. e.g. The  version below is still `2.8.1`, the `Build Status` above is not working anymore. It could be a separate PR though.

**Review by blakeli0 — APPROVED** [review]

File: gax-java/README.md (hunk 6 above)
Comment (suztomo) [reply gax-java/README.md:19]: Updated to the latest version and removed the build status badge.
